src/scmdata/run.py

```diff
@@ -638,61 +637,56 @@ def _indent(s):
             len(self), len(self.time_points), time_str, meta_str
         )
 
-    @staticmethod
     def _binary_op(
-        f: Callable[..., Any], reflexive=False, **kwargs,
+        self, other, f, reflexive=False, **kwargs,
     ) -> Callable[..., "ScmRun"]:
-        @functools.wraps(f)
-        def func(self, other):
-            if isinstance(other, ScmRun):
-                return NotImplemented
-
-            is_scalar = isinstance(other, (numbers.Number, pint.Quantity))
-            if not is_scalar:
-                other_ndim = len(other.shape)
-                if other_ndim == 1:
-                    if other.shape[0] != self.shape[1]:
-                        raise ValueError(
-                            "only vectors with the same number of timesteps "
-                            "as self ({}) are supported".format(self.shape[1])
-                        )
-                else:
+        if isinstance(other, ScmRun):
+            return NotImplemented
+
+        is_scalar = isinstance(other, (numbers.Number, pint.Quantity))
+        if not is_scalar:
+            other_ndim = len(other.shape)
+            if other_ndim == 1:
+                if other.shape[0] != self.shape[1]:
                     raise ValueError(
-                        "operations with {}d data are not supported".format(other_ndim)
+                        "only vectors with the same number of timesteps "
+                        "as self ({}) are supported".format(self.shape[1])
                     )
+            else:
+                raise ValueError(
+                    "operations with {}d data are not supported".format(other_ndim)
+                )
 
-            def _perform_op(df):
-                if isinstance(other, pint.Quantity):
-                    try:
-                        data = df.values * ur(df.get_unique_meta("unit", True))
-                        use_pint = True
-                    except KeyError:
-                        # let Pint assume dimensionless and raise an error as
-                        # necessary
-                        data = df.values
-                        use_pint = False
-                else:
+        def _perform_op(df):
+            if isinstance(other, pint.Quantity):
+                try:
+                    data = df.values * ur(df.get_unique_meta("unit", True))
+                    use_pint = True
+                except KeyError:
+                    # let Pint assume dimensionless and raise an error as
```

Comment:
I put this line in there and I'm not actually sure it's a good idea anymore (especially if I didn't bother testing it...). I'd be tempted to just put the following here (and below where it's copied)

```python
except KeyError:  # pragma: no cover # emergency valve
    raise KeyError("No `unit` column in your metadata, cannot perform operations with pint quantities")
```

Reply:
Fixed it for ScmRun as it is rather difficult to create an object which doesn't have units, but there is existing test coverage for Timeseries with no units. If it really should change it can happen via a different PR